attribute/value.go

```diff
@@ -66,11 +67,11 @@ func BoolValue(v bool) Value {
 
 // BoolSliceValue creates a BOOLSLICE Value.
 func BoolSliceValue(v []bool) Value {
-	cp := make([]bool, len(v))
-	copy(cp, v)
+	cp := reflect.New(reflect.ArrayOf(len(v), reflect.TypeOf(false)))
+	copy(cp.Elem().Slice(0, len(v)).Interface().([]bool), v)
 	return Value{
 		vtype: BOOLSLICE,
-		slice: &cp,
+		slice: cp.Elem().Interface(),
 	}
 }
```

Comment:
The `bool`, `int64`, `float64`, and `string` (`int` is unique here) value functions can be abstracted to use the same conversion function:

```go
func sliceValue[T bool | int64 | float64 | string](v []T) any {
	var zero T
	cp := reflect.New(reflect.ArrayOf(len(v), reflect.TypeOf(zero)))
	copy(cp.Elem().Slice(0, len(v)).Interface().([]T), v)
	return cp.Elem().Interface()
}
```

This then becomes:

```go
func BoolSliceValue(v []bool) Value {
	return Value{vtype: BOOLSLICE, slice: sliceValue(v)}
}
```

Similar to the other conversion functions, this new `sliceValue` function could be added to the `internal` package.

Reply:
Agree, looks nice @MrAlias.